### PR TITLE
8477 task(style): re-skin footer

### PR DESCRIPTION
**Pull request by skibinska**
Relates https://github.com/wellcometrust/corporate/issues/8477 and https://github.com/wellcometrust/corporate/issues/8352

### Context

Re-skin to match header UI and visual design

Designs are on Sketch Cloud here
https://www.sketch.com/s/42765e65-a4b1-4c1b-bd45-9d0688b1c3f2

### This PR

- removes `<Footer>` component
- updates `<NewsletterForm>` style to match new footer design

Updated footer:

Screenshot 2021-06-16 at 09 55 15

### Test

(I tested it on Chrome, Firefox, Safari and Chrome on Android)

- go to https://qa-8554-yellow-links.wellcome-corporate-fe.org.uk/

### CODE REVIEW

**Review by chriddell — COMMENTED** [review]
Before I go into the weeds of the code, have we/you considered moving the Footer component to corporate-react?

My reasoning being that it's a highly specific component, unlikely to be used elsewhere on other Wellcome products.

I'll take a look at the code in the PR, now, anyway.

**Comment by creido-welly** [comment]
Echoing what @chriddell said above, there is a deprecation notice to effectively move this into CR. https://github.com/wellcometrust/corporate/issues/8352

**Comment by skibinska** [comment]
Sure, I will move it to CR.

**Comment by creido-welly** [comment]
Can I also suggest it is renamed to SiteFooter since Header was renamed to SiteHeader when that was moved across

**Comment by chriddell** [comment]
@skibinska would you prefer to close this PR and open a new PR in corporate-react with this code? 

Or would you like us to review this PR, merge it in, create a new issue for moving the footer, and then open a new PR against corporate-react?

I don't think it matters either way so I'll leave it up to you. The first option will probs be quicker.

**Comment by skibinska** [comment]
@chriddell I will open a new PR in CR.

File: src/components/NewsletterForm/NewsletterFormFooter.tsx
Comment (creido-welly) [inline src/components/NewsletterForm/NewsletterFormFooter.tsx]: This should probably be Dotdigital (the new name for Dotmailer)

File: src/components/NewsletterForm/NewsletterFormFooter.tsx
Comment (creido-welly) [inline src/components/NewsletterForm/NewsletterFormFooter.tsx]: Dotmailer should be Dotdigital

**Review by creido-welly — APPROVED** [review]
Aside from the Dotdigital typos I think this is ok. It might be worth holding off from publishing the package until you're certain the new site footer is fully in place in CR.

**Review by creido-welly — APPROVED** [review]

**Pull request merged**
